File: Firebase/creative_api/functions/src/Service/RouteService.js
const Route = require("./../Class/Route");
const routeRepository = require("./../Repository/RouteRepository");

class RouteSevice {
    async saveRoute(route) {
        return (await routeRepository.save(route.get()));
    }
    async getById(id) {
        return (await routeRepository.find(id));
    }
    async getAll() {
        return (await routeRepository.findAll());
    }
    async updateRoute(route){
        return (await routeRepository.update(route.get()));
    }
}

module.exports = new RouteSevice();

File: C#/Truckleer/Truckleer.Creative/Modules/Route/RouteService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Truckleer.Modules
{
    class RouteService
    {
        //Route repository property
        readonly RouteRepository repository;
        //Class constructor
        public RouteService()
        {
            //initializate route repository property
            repository = new RouteRepository();
        }
        //Method to findAll routes
        public List<Route> FindAll()
        {
            return repository.FindAll().Result;
        }
        //Method to find one specific route by id
        public Route FindOne(string Id)
        {
            return repository.Find(Id).Result;
        }
        //Method to Save or Update route
        public bool Save(Route route)
        {
            return repository.Save(route).Result;
        }
    }
}
